Building Avalonia.Desktop.slnf from source fails with NuGet errors due to missing <packageSourceMapping> element

### Describe the bug

Building `Avalonia.Desktop.slnf` from source causes numerous NuGet restore errors if global `NuGet.Config` (`%APPDATA%\NuGet\NuGet.Config` in Windows) contains `<packageSourceMapping>` configuration which is incompatible with Avalonia's added `api.nuget.org` element.

### To Reproduce

1. Have a global `NuGet.Config` with `<packageSourceMapping>` which doesn't include `api.nuget.org`. For example:

    ```xml
    <?xml version="1.0" encoding="utf-8"?>
    <configuration>
      <packageSources>
        <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
      </packageSources>
      <packageRestore>
        <add key="enabled" value="True" />
        <add key="automatic" value="True" />
      </packageRestore>
      <bindingRedirects>
        <add key="skip" value="False" />
      </bindingRedirects>
      <packageManagement>
        <add key="format" value="1" />
        <add key="disabled" value="True" />
      </packageManagement>
      <packageSourceMapping>
        <packageSource key="nuget.org">
          <package pattern="*" />
        </packageSource>
      </packageSourceMapping>
    </configuration>
    ```

1. Clone Avalonia master branch.

1. Run `dotnet build Avalonia.Desktop.slnf`

Build fails. In addition to references to recently renamed/removed projects, there're numerous NuGet restore errors:

> C:\Docs\Projects\Avalonia\tests\Avalonia.Markup.Xaml.UnitTests\Avalonia.Markup.Xaml.UnitTests.csproj : error NU1100: Unable to resolve 'Moq (>= 4.20.72)' for 'net10.0'. PackageSourceMapping is enabled, the following source(s) were not considered: api.nuget.org, C:\Program Files\dotnet\library-packs. [C:\Docs\Projects\Avalonia\Avalonia.Desktop.slnf]  
> C:\Docs\Projects\Avalonia\tests\Avalonia.Skia.RenderTests\Avalonia.Skia.RenderTests.csproj : error NU1100: Unable to resolve 'Moq (>= 4.20.72)' for 'net10.0'. PackageSourceMapping is enabled, the following source(s) were not considered: api.nuget.org, C:\Program Files\dotnet\library-packs. [C:\Docs\Projects\Avalonia\Avalonia.Desktop.slnf]  
> C:\Docs\Projects\Avalonia\tests\Avalonia.RenderTests.WpfCompare\Avalonia.RenderTests.WpfCompare.csproj : error NU1100: Unable to resolve 'Xunit.StaFact (>= 1.2.46-alpha)' for 'net10.0-windows7.0'. PackageSourceMapping is enabled, the following source(s) were not considered: api.nuget.org, C:\Program Files\dotnet\library-packs. [C:\Docs\Projects\Avalonia\Avalonia.Desktop.slnf]  
> C:\Docs\Projects\Avalonia\tests\Avalonia.Markup.Xaml.UnitTests\Avalonia.Markup.Xaml.UnitTests.csproj : error NU1100: Unable to resolve 'xunit (>= 2.9.2)' for 'net10.0'. PackageSourceMapping is enabled, the following source(s) were not considered: api.nuget.org, C:\Program Files\dotnet\library-packs. [C:\Docs\Projects\Avalonia\Avalonia.Desktop.slnf]  
> C:\Docs\Projects\Avalonia\tests\Avalonia.RenderTests.WpfCompare\Avalonia.RenderTests.WpfCompare.csproj : error NU1100: Unable to resolve 'Moq (>= 4.20.72)' for 'net10.0-windows7.0'. PackageSourceMapping is enabled, the following source(s) were not considered: api.nuget.org, C:\Program Files\dotnet\library-packs. [C:\Docs\Projects\Avalonia\Avalonia.Desktop.slnf]  
> C:\Docs\Projects\Avalonia\tests\Avalonia.Markup.Xaml.UnitTests\Avalonia.Markup.Xaml.UnitTests.csproj : error NU1100: Unable to resolve 'xunit.assert (>= 2.9.2)' for 'net10.0'. PackageSourceMapping is enabled, the following source(s) were not considered: api.nuget.org, C:\Program Files\dotnet\library-packs. [C:\Docs\Projects\Avalonia\Avalonia.Desktop.slnf]  
> C:\Docs\Projects\Avalonia\tests\Avalonia.RenderTests.WpfCompare\Avalonia.RenderTests.WpfCompare.csproj : error NU1100: Unable to resolve 'xunit (>= 2.9.2)' for 'net10.0-windows7.0'. PackageSourceMapping is enabled, the following source(s) were not considered: api.nuget.org, C:\Program Files\dotnet\library-packs. [C:\Docs\Projects\Avalonia\Avalonia.Desktop.slnf]  
> C:\Docs\Projects\Avalonia\tests\Avalonia.Markup.Xaml.UnitTests\Avalonia.Markup.Xaml.UnitTests.csproj : error NU1100: Unable to resolve 'xunit.core (>= 2.9.2)' for 'net10.0'. PackageSourceMapping is enabled, the following source(s) were not considered: api.nuget.org, C:\Program Files\dotnet\library-packs. [C:\Docs\Projects\Avalonia\Avalonia.Desktop.slnf]  

### Expected behavior

Solution filter is successfully built.

### Avalonia version

12

### OS

Windows

### Additional context

When clearing `<packageSources>`, `<packageSourceMapping>` and other elements should be cleared too to avoid conflicts. For example, here's an example from NuGet's own NuGet.Config:

https://github.com/NuGet/NuGet.Client/blob/dev/NuGet.Config

Note that it clears all elements: `packageSourceMapping`, `auditSources`, `disabledPackageSources`, `fallbackPackageFolders`.

--- a/NuGet.Config
+++ b/NuGet.Config
@@ -4,5 +4,18 @@
   <packageSources>
     <clear />
     <add key="api.nuget.org" value="https://api.nuget.org/v3/index.json" />
+    <add key="azure-dotnet10-transport" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet10-transport/nuget/v3/index.json" />
   </packageSources>
+  <packageSourceMapping>
+    <clear />
+  </packageSourceMapping>
+  <auditSources>
+    <clear />
+  </auditSources>
+  <disabledPackageSources>
+    <clear />
+  </disabledPackageSources>
+  <fallbackPackageFolders>
+    <clear />
+  </fallbackPackageFolders>
 </configuration>

--- a/nukebuild/_build.csproj
+++ b/nukebuild/_build.csproj
@@ -9,8 +9,6 @@
     <TargetFramework>$(AvsCurrentTargetFramework)</TargetFramework>
     <!-- See https://github.com/nuke-build/nuke/issues/818 -->
     <EnableUnsafeBinaryFormatterSerialization>true</EnableUnsafeBinaryFormatterSerialization>
-    <!-- Necessary for Microsoft.DotNet.ApiDiff.Tool -->
-    <RestoreAdditionalProjectSources>https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet10-transport/nuget/v3/index.json</RestoreAdditionalProjectSources>
   </PropertyGroup>
 
   <Import Project="..\build\JetBrains.dotMemoryUnit.props" />

